meshroom/ui/qml/Controls/SearchBar.qml

```diff
@@ -35,10 +58,45 @@ FocusScope {
             Layout.fillWidth: true
             selectByMouse: true
 
+            rightPadding: clear.width
+
+            // The text field is visible either when toggle is not activated or the visible property is set
+            visible: root.toggle ? root.isVisible : true
+
             // ensure the field has focus when the text is modified
             onTextChanged: {
                 forceActiveFocus()
             }
+
+            // Handle enter Key press and forward it to the parent
+            Keys.onPressed: (event)=> {
+                if ((event.key == Qt.Key_Return)) {
+                    event.accepted = true
+                    root.returnPressed()
+                }
+            }
+
+            MaterialToolButton {
+                id: clear
+
+                // Anchors
+                anchors.right: parent.right
+                anchors.verticalCenter: parent.verticalCenter
```

Comment:
I think some minor right-padding should be added: when the "clear" button appears, the search bar generally has the focus (so is "highlighted") and when hovering over the button, it goes over the highlighting, which is not very pleasing.
![Screenshot_2024-11-12_10-28-03](https://github.com/user-attachments/assets/6353a332-0cf0-46d1-9378-5bd68e89aa33)

Reply:
Yes, this is now fixed with the change here, https://github.com/alicevision/Meshroom/blob/1a0d2cf905903091012db3384c985a1d198490c8/meshroom/ui/qml/Controls/SearchBar.qml#L83
This change ensures we're leaving sufficient space on the right side as to not overlap with the parent's bounds.
Hoping this makes it better 🙂